ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-isis.j2

```diff
@@ -83,6 +188,75 @@ router isis {{ router_isis.instance }}
 {%         if router_isis.graceful_restart.restart_hold_time is arista.avd.defined %}
    graceful-restart restart-hold-time {{ router_isis.graceful_restart.restart_hold_time }}
 {%         endif %}
+{%     endif %}
+{%     if router_isis.authentication is arista.avd.defined %}
+{%         if router_isis.authentication.both is arista.avd.defined %}
+{%             if router_isis.authentication.both.key_id is arista.avd.defined %}
+{%                 for auth_key_id in router_isis.authentication.both.key_id %}
+{%                     if auth_key_id.id is arista.avd.defined and auth_key_id.algorithm is arista.avd.defined %}
+{%                         if auth_key_id.key_type is arista.avd.defined and auth_key_id.key is arista.avd.defined %}
+{%                             if auth_key_id.rfc_5310 is arista.avd.defined(true) %}
+   authentication key-id {{ auth_key_id.id }} algorithm {{ auth_key_id.algorithm }} rfc-5310 key {{ auth_key_id.key_type }} {{ auth_key_id.key }}
+{%                             else %}
+   authentication key-id {{ auth_key_id.id }} algorithm {{ auth_key_id.algorithm }} key {{ auth_key_id.key_type }} {{ auth_key_id.key }}
+{%                             endif %}
+{%                         endif %}
+{%                     endif %}
+{%                 endfor %}
+{%             endif %}
+{%         endif %}
+{%         if router_isis.authentication.level_1 is arista.avd.defined %}
+{%             if router_isis.authentication.level_1.key_id is arista.avd.defined %}
+{%                 set both_key_id = [] %}
```

Comment:
Please add the key ids to a list during parsing above.

Reply:
Fixed